core/src/main/java/org/apache/gravitino/authorization/OwnershipManager.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.gravitino.authorization;
+
+import java.io.IOException;
+import java.util.List;
+import org.apache.gravitino.Entity;
+import org.apache.gravitino.EntityStore;
+import org.apache.gravitino.NameIdentifier;
+import org.apache.gravitino.exceptions.NoSuchEntityException;
+import org.apache.gravitino.exceptions.NotFoundException;
+import org.apache.gravitino.exceptions.OwnerNotFoundException;
+import org.apache.gravitino.meta.GroupEntity;
+import org.apache.gravitino.meta.UserEntity;
+import org.apache.gravitino.relation.Relation;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** OwnershipManager is used for manage the owner of securable objects, user, group and role. */
+public class OwnershipManager {
+  private static final Logger LOG = LoggerFactory.getLogger(OwnershipManager.class);
+  private final EntityStore store;
+
+  public OwnershipManager(EntityStore store) {
+    this.store = store;
+  }
+
+  public void setOwner(
+      NameIdentifier identifier, Entity.EntityType type, String ownerName, Owner.Type ownerType) {
```

Comment:
Owner is binding to the metadata/securable object, from interface level you should have such concept.

Reply:
I changed to meta data object.